File: backend/app/schemas/review.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class ReviewDecisionCreate(BaseModel):
    decision: str = Field(..., description="UNDER_REVIEW, CLEARED, ESCALATED, NEEDS_EVIDENCE")
    notes: Optional[str] = None


class ReviewDecisionResponse(BaseModel):
    id: str
    contract_id: str
    reviewer_id: Optional[str] = None
    reviewer_name: Optional[str] = None
    decision: str
    notes: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/review.ts
export type ReviewDecisionType = 'UNDER_REVIEW' | 'CLEARED' | 'ESCALATED' | 'NEEDS_EVIDENCE';

export interface ReviewDecision {
  id: string;
  contract_id: string;
  reviewer_id?: string | null;
  reviewer_name?: string | null;
  decision: ReviewDecisionType;
  notes?: string | null;
  created_at: string;
}

export interface ReviewDecisionPayload {
  decision: ReviewDecisionType;
  notes?: string;
}

export interface Alert {
  id: string;
  contract_id: string;
  contract_number?: string | null;
  contract_title?: string | null;
  risk_level: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
  message: string;
  read: boolean;
  created_at: string;
}

export interface UserProfile {
  id: string;
  email: string;
  full_name: string | null;
  role: 'ADMIN' | 'AUDITOR';
  department: string | null;
}
